presto-parquet/src/main/java/com/facebook/presto/parquet/reader/ColumnIndexStoreImpl.java

```diff
@@ -0,0 +1,165 @@
+/*
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package com.facebook.presto.parquet.reader;
+
+import com.facebook.presto.parquet.ParquetDataSource;
+import org.apache.parquet.hadoop.metadata.BlockMetaData;
+import org.apache.parquet.hadoop.metadata.ColumnChunkMetaData;
+import org.apache.parquet.hadoop.metadata.ColumnPath;
+import org.apache.parquet.internal.column.columnindex.ColumnIndex;
+import org.apache.parquet.internal.column.columnindex.OffsetIndex;
+import org.apache.parquet.internal.filter2.columnindex.ColumnIndexStore;
+
+import java.io.IOException;
+import java.util.HashMap;
+import java.util.Map;
+import java.util.Set;
+
+import static java.util.Collections.emptySet;
+
+/**
+ * Internal implementation of {@link ColumnIndexStore}.
+ */
+public class ColumnIndexStoreImpl
+        implements ColumnIndexStore
+{
+    private interface IndexStore
+    {
+        ColumnIndex getColumnIndex();
+
+        OffsetIndex getOffsetIndex();
+    }
+
+    private class IndexStoreImpl
+            implements IndexStore
+    {
+        private final ColumnChunkMetaData meta;
+        private ColumnIndex columnIndex;
+        private boolean columnIndexRead;
+        private final OffsetIndex offsetIndex;
+
+        IndexStoreImpl(ColumnChunkMetaData meta)
+        {
+            this.meta = meta;
+            OffsetIndex oi;
```

Comment:
s/oi/offsetIndex/g

Reply:
refactor a bit of the code and oi is gone now.